app/controllers/users_controller.rb

```diff
@@ -0,0 +1,24 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      redirect_to user_path(@user)
+    else
+      flash[:error] = 'E-mail address must be unique.'
```

Comment:
Leaving a comment for us to come back for some potential future refactor by looking into model error message handling.